jetty-webapp/src/test/java/org/eclipse/jetty/webapp/ServletContextTmpAttributeTest.java

```diff
@@ -0,0 +1,176 @@
+        File temp = webAppContext.getTempDirectory();
+        assertThat("webAppContext is the temp directory created",
+                   temp.toPath(),
+                   is(tmp));
+    }
+
+    /**
+     * ServletContext.TEMPDIR has <code>""</code> value
+     * IllegalStateException
+     */
+    @Test
+    public void attributeWithEmptyStringValue() throws Exception
+    {
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        WebAppContext webAppContext = new WebAppContext();
+        webAppContext.setAttribute(ServletContext.TEMPDIR, "");
+        assertThrows(IllegalStateException.class, () -> webInfConfiguration.resolveTempDirectory(webAppContext));
+    }
+
+    /**
+     * ServletContext.TEMPDIR as File to valid directory
+     */
+    @Test
+    public void attributeWithValidFileDirectoryValue() throws Exception
+    {
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        WebAppContext webAppContext = new WebAppContext();
+        Path tmp = Files.createTempDirectory("jetty_test");
+        webAppContext.setAttribute(ServletContext.TEMPDIR, tmp.toFile());
+        webInfConfiguration.resolveTempDirectory(webAppContext);
+        File temp = webAppContext.getTempDirectory();
+        assertThat("webAppContext is the temp directory created",
+                   temp.toPath(),
+                   is(tmp));
+    }
+
+    /**
+     * ServletContext.TEMPDIR as Path to valid directory
+     */
+    @Test
+    public void attributeWithValidPathDirectoryValue() throws Exception
+    {
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        WebAppContext webAppContext = new WebAppContext();
+        Path tmp = Files.createTempDirectory("jetty_test");
+        webAppContext.setAttribute(ServletContext.TEMPDIR, tmp);
+        // FIXME we should have an exception here
+        webInfConfiguration.resolveTempDirectory(webAppContext);
+    }
+
+    /**
+     * ServletContext.TEMPDIR has invalid <code>String</code> directory value
+     * IllegalStateException
+     */
+    @Test
+    public void attributeWithInvalidStringValue() throws Exception
+    {
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
+        WebAppContext webAppContext = new WebAppContext();
+        webAppContext.setAttribute(ServletContext.TEMPDIR, "/French/Cheese/Rocks");
+        assertThrows(IllegalStateException.class, () -> webInfConfiguration.resolveTempDirectory(webAppContext));
+    }
+
+    /**
+     * ServletContext.TEMPDIR has invalid <code>String</code> directory value (wrong permission to write into it)
+     * IllegalStateException
+     */
+    @Disabled("will fail if executed as root or super power user so Disabled it")
+    public void attributeWithInvalidPermissionStringValue() throws Exception
+    {
+        WebInfConfiguration webInfConfiguration = new WebInfConfiguration();
```

Comment:
Couldn't the test create a directory, then remove user read permission from it?

Reply:
but then the current user/process won't be able to delete it

Reply:
Once you've removed write permission, can't you put it back again before the test ends?